ethereum/executionclient/src/main/java/tech/pegasys/teku/ethereum/executionclient/methods/EngineForkChoiceUpdatedV2.java

```diff
@@ -0,0 +1,93 @@
+ */
+
+package tech.pegasys.teku.ethereum.executionclient.methods;
+
+import java.util.Optional;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import tech.pegasys.teku.ethereum.executionclient.ExecutionEngineClient;
+import tech.pegasys.teku.ethereum.executionclient.response.ResponseUnwrapper;
+import tech.pegasys.teku.ethereum.executionclient.schema.ForkChoiceStateV1;
+import tech.pegasys.teku.ethereum.executionclient.schema.ForkChoiceUpdatedResult;
+import tech.pegasys.teku.ethereum.executionclient.schema.PayloadAttributesV1;
+import tech.pegasys.teku.ethereum.executionclient.schema.PayloadAttributesV2;
+import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.SpecMilestone;
+import tech.pegasys.teku.spec.executionlayer.ForkChoiceState;
+import tech.pegasys.teku.spec.executionlayer.PayloadBuildingAttributes;
+
+public class EngineForkChoiceUpdatedV2
+    extends AbstractEngineJsonRpcMethod<
+        tech.pegasys.teku.spec.executionlayer.ForkChoiceUpdatedResult> {
+
+  private static final Logger LOG = LogManager.getLogger();
+  private final Spec spec;
+
+  public EngineForkChoiceUpdatedV2(
+      final ExecutionEngineClient executionEngineClient, final Spec spec) {
+    super(executionEngineClient);
+    this.spec = spec;
+  }
+
+  @Override
+  public String getName() {
+    return "engine_forkChoiceUpdated";
+  }
+
+  @Override
+  public int getVersion() {
+    return 2;
+  }
+
+  @Override
+  public SafeFuture<tech.pegasys.teku.spec.executionlayer.ForkChoiceUpdatedResult> execute(
+      final JsonRpcRequestParams params) {
+    final ForkChoiceState forkChoiceState = params.getRequiredParameter(0, ForkChoiceState.class);
+    final Optional<PayloadBuildingAttributes> payloadBuildingAttributes =
+        params.getOptionalParameter(1, PayloadBuildingAttributes.class);
+
+    LOG.trace(
+        "calling engineForkChoiceUpdatedV1(forkChoiceState={}, payloadAttributes={})",
+        forkChoiceState,
+        payloadBuildingAttributes);
+
+    final Optional<PayloadAttributesV1> maybePayloadAttributes =
+        payloadBuildingAttributes.flatMap(
+            attributes ->
+                spec.atSlot(attributes.getBlockSlot())
+                        .getMilestone()
+                        .isGreaterThanOrEqualTo(SpecMilestone.CAPELLA)
+                    ? PayloadAttributesV2.fromInternalPayloadBuildingAttributesV2(
+                        payloadBuildingAttributes)
+                    : PayloadAttributesV1.fromInternalPayloadBuildingAttributes(
+                        payloadBuildingAttributes));
+    LOG.trace(
+        "calling engineForkChoiceUpdatedV2(forkChoiceState={}, payloadAttributes={})",
+        forkChoiceState,
+        maybePayloadAttributes);
+    return executionEngineClient
+        .forkChoiceUpdatedV2(
+            ForkChoiceStateV1.fromInternalForkChoiceState(forkChoiceState), maybePayloadAttributes)
```

Comment:
i haven't reconciled this, but can we check? v1 v2... mixy...

Reply:
This one is correct. We only have `ForkChoiceStateV1`. 

Snippet from `ExecutionEngineClient` interface:
```
  SafeFuture<Response<ForkChoiceUpdatedResult>> forkChoiceUpdatedV1(
      ForkChoiceStateV1 forkChoiceState, Optional<PayloadAttributesV1> payloadAttributes);

  SafeFuture<Response<ForkChoiceUpdatedResult>> forkChoiceUpdatedV2(
      ForkChoiceStateV1 forkChoiceState, Optional<PayloadAttributesV1> payloadAttributes);
```